project/src/main/webapp/keyword.js

```diff
@@ -0,0 +1,31 @@
+function editedKeywordExtraction(userInput) {
```

Comment:
Please add comments for each function and give them more descriptive names (e.g. datamuseKeyWordExtraction vs naiveKeywordExtraction or maybe getQuestionType is more appropriate?)